Profiler/utils/Debug.h

```diff
@@ -9,32 +9,16 @@
 class Debug
 {
 public:
+	Debug();
+
 	/** Logs the given message to the debug log. */
-	static void log(std::string message);
+	void log(std::string message);
 
-	/** Logs the current stack trace. */
-	static void logStacktrace(std::string context);
+	/** Logs the given context and the current stacktrace to the debug log. */
+	void logErrorWithStracktrace(std::string context);
 
 	virtual ~Debug();
-
 private:
-	Debug();
-
-	void logInternal(std::string message);
-
-	/**
-	 * Implements a thread-safe singleton pattern and ensures that the instance
-	 * is properly destroyed and thus the file handle freed.
-	 * See https://stackoverflow.com/a/1008289/1396068
-	 */
-	static Debug& getInstance()
```

Comment:
The idea behind making this a singleton was that you can call the log from any part of our code (e.g. utils functions) and have a thread-safe log output. Right now, I either have to explicitly pass the debugLog instance around in the code if I want to debug something or I have to create a new instance, which is not thread safe, i.e. it's possible to get mangled log output with overlapping log messages. I'd prefer to have this as a singleton again